script/tool/lib/src/repo_package_info_check_command.dart

```diff
@@ -115,92 +135,110 @@ class RepoPackageInfoCheckCommand extends PackageLoopingCommand {
       errors.add('Missing CODEOWNERS entry');
     }
 
+    // All packages should have an auto-applied label. For plugins, only the
+    // group needs a rule, so check the app-facing package.
+    if (!(package.isFederated && !package.isAppFacing) &&
+        !(_autoLabeledPackages.contains(packageName))) {
+      printError('${indentation}Missing a rule in .github/labeler.yml.');
+      errors.add('Missing auto-labeler entry');
+    }
+
     // The content of ci_config.yaml must be valid if there is one.
     if (package.ciConfigFile.existsSync()) {
       errors.addAll(
           _validateCiConfig(package.ciConfigFile, mainPackage: package));
     }
 
-    // Any published package should be in the README table.
+    // All published packages should have a README.md entry.
+    if (package.isPublishable()) {
+      errors.addAll(_validateRootReadme(package));
+    }
+
+    return errors.isEmpty
+        ? PackageResult.success()
+        : PackageResult.fail(errors);
+  }
+
+  List<String> _validateRootReadme(RepositoryPackage package) {
+    final List<String> errors = <String>[];
+
     // For federated plugins, only the app-facing package is listed.
-    if (package.isPublishable() &&
-        (!package.isFederated || package.isAppFacing)) {
-      final List<String>? cells = _readmeTableEntries[packageName];
+    if (package.isFederated && !package.isAppFacing) {
```

Comment:
This all looks like a large diff because I extracted the logic into a helper and switched the if-wrapper to an early return, changing the nesting. Essentially the entire body of this method is just a move of the code that used to be inlined in `runForPackage`.